src/test/java/uk/gov/pay/api/it/CreateAgreementIT.java

```diff
@@ -39,17 +41,29 @@ public void shouldCreateAgreement() {
     }
 
     @Test
-    public void shouldReturn422WhenWhenReferenceIsEmptyString() {
+    public void shouldReturn400WhenWhenReferenceIsEmptyString() {
         publicAuthMockClient.mapBearerTokenToAccountId(API_KEY, GATEWAY_ACCOUNT_ID, CARD);
         CreateAgreementRequestParams createAgreementRequestParams = aCreateAgreementRequestParams()
                 .withReference("")
                 .build();
         postAgreementRequest(agreementPayload(createAgreementRequestParams))
-                .statusCode(HttpStatus.SC_UNPROCESSABLE_ENTITY)
+                .statusCode(HttpStatus.SC_BAD_REQUEST)
```

Comment:
This seems to be equivalent to what happens with creating payments, which is good.